CONTRIBUTING.md

```diff
@@ -0,0 +1,216 @@
+
+* On Windows, we recommend [git Bash](https://gitforwindows.org/) rather than the [official Windows version of git](https://git-scm.com/download/win), because git Bash provides its own shell that includes many Linux-equivalent command line programs that are useful for development. Windows 10 also offers the [Windows subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/about) that offers similar functionality to git Bash.
+
+> **Git GUI alternative**
+> [GitHub desktop](https://desktop.github.com/) is a GUI alternative to command line git that some users appreciate; it is available for :+1: Windowns and MacOS
+
+Once git is installed, the only absolutely necessary configuration step is identifying yourself and your contact info:
+
+```bash
+~$ git config --global user.name "Your Name"
+~$ git config --global user.email you@yourdomain.example.com
+```
+
+Make sure that the same email address is associated with your GitHub account and with your local git configuration. It is possible to associate multiple emails with a GitHub account, so if you initially set them up with different emails, just add the local email to the GitHub account.
+
+To customize git’s behavior see [configuring git](https://www.git-scm.com/book/en/v2/Customizing-Git-Git-Configuration) for more information. Once you have git installed and configured, go to [the Neurosym-lab/netpyne GitHub page](https://github.com/Neurosim-lab/netpyne) and create a [fork](https://help.github.com/en/articles/fork-a-repo) into your GitHub user account.
+
+### Make a local copy of GitHub repository
+
+> **Supported Python environments**
+>We strongly recommend the [Anaconda](https://www.anaconda.com/distribution/) or [Miniconda](https://conda.io/en/latest/miniconda.html) environment managers for Python.
+
+You must clone the Nerosim-lab/netpyne repository from your remote fork, and also connect the local copy to the `upstream` version of the codebase, so you can stay up-to-date with changes from other contributors.
+
+Make a local clone of your remote fork (`origin`):
+
+```bash
+ ~$ cd $INSTALL_LOCATION
+ ~$ git clone https://github.com/$GITHUB_USERNAME/netpyne.git
+```
+
+> **Remote URLs in git**
+>Here we use `git://` instead of `https://` in the URL for the `upstream` remote repository. `git://` URLs are read-only, so you can `pull` changes from `upstream` into your local copy (to stay up-to-date with changes from other contributors) but you cannot `push` changes from your computer into the `upstream` remote. Instead, you must `push` your changes to your own remote fork (`origin`) first, and then create a pull request from your remote into the `upstream` remote. In [a later section](#connecting-to-github-with-ssh-optional) you’ll see a third kind of remote URL for connecting to GitHub using SSH.

```

Comment:
sorry, confused ... where is git:// used? in the code above it uses https

Reply:
:+1: I replace for:
> **Remote URLs in git**
>Here we use `https://`, instead of `git@` addresses, in the URL for the `upstream` remote repository. `https://` URLs are read-only, so you can `pull` changes from `upstream` into your local copy (to stay up-to-date with changes from other contributors) but you cannot `push` changes from your computer into the `upstream` remote. Instead, you must `push` your changes to your own remote fork (`origin`) first, and then create a pull request from your remote into the `upstream` remote. In [a later section](#connecting-to-github-with-ssh-optional) you’ll see a second kind of remote URL for connecting to GitHub using SSH.